ctgan/synthesizers/ctgan.py

```diff
@@ -420,6 +420,7 @@ def fit(self, train_data, discrete_columns=(), epochs=None):
                       f'Loss D: {loss_d.detach().cpu(): .4f}',
                       flush=True)
 
+    @random_state
```

Comment:
Should we also set the `random_state` to the `fit` method?
This would serve a differen purpose than the one that originally motivates this PR, but it may be very inconsistent to if the cycle of setting the seed, fitting and sampling can produce different results when repeated.